odo/numpy_dtype.py

```diff
@@ -23,6 +25,9 @@ def unit_to_dtype(ds):
     if isinstance(ds, DataShape):
         ds = ds.measure
     if isinstance(ds, Option) and isscalar(ds) and isnumeric(ds):
+        with ignoring(NameError):
+            if isinstance(ds.ty, Decimal):
+                return unit_to_dtype(str(ds.ty.to_numpy_dtype()).replace('int', 'float'))
```

Comment:
I'll try to confirm with a concrete example later today or tomorrow, but the existing code stringifies the subtype of the Option and then replaces int with float, but if the type is `Option(Decimal)`, this code won't apply, it won't strip off the Option type, and will end up triggering the Option `to_numpy_dtype` error.

I'll try to verify this when I'm not on my phone. :) I believe I encountered this in testing after adding `Decimal`.